contrib/automation_tests/test_cases/capture_window.py

```diff
@@ -167,15 +182,58 @@ def _execute(self, expected_count: int = None, expected_names: List[str or Itera
         if expected_names and not allow_additional_tracks:
             self.expect_eq(names_found, expected_count, "No additional tracks are found")
 
-    @staticmethod
-    def _match(expected_name: str or Iterable[str], found_name: str) -> bool:
-        if isinstance(expected_name, str):
-            return fnmatch(found_name, expected_name)
-        else:
-            for option in expected_name:
-                if fnmatch(found_name, option):
-                    return True
-        return False
+
+class CollapsingTrackBase(MatchTracksBase):
+    """
+    Clicks on a track's triangle toggle and calls compares the track's height by calling `_verify_height`.
+    """
+
+    def _execute(self, expected_name: str):
+        """
+        :param expected_name: The exact name of the track to be collapsed. "*" is allowed as placeholder.
+        """
+        assert (expected_name is not "")
+
+        tracks = self._find_tracks()
+
+        found_track = False
+        for track in [Track(t) for t in tracks]:
+            track_name = track.name
+            if self._match(expected_name, track_name):
```

Comment:
`self._find_tracks()` has a filter parameter that you can use to only return tracks matching a certain pattern, equal to what's checked in `self._match`. I'd recomment to use this and remove the inheritance from `MatchTracksBase`.

Reply:
Ah thanks. Done. I also deleted the MatchTracksBase class.